backend/typescript/services/implementations/reviewService.ts

```diff
@@ -399,6 +480,266 @@ class ReviewService implements IReviewService {
 
     return result;
   }
+
+  async updateReviews(id: string, entity: ReviewRequestDTO): Promise<void> {
+    try {
+      await this.db.transaction(async (t) => {
+        const reviewToUpdate = await PgReview.findByPk(id, {
+          transaction: t,
+          include: [{ all: true, nested: true }],
+        });
+
+        if (!reviewToUpdate) {
+          throw new Error(`Review id ${id} not found`);
+        }
+
+        // delete tags
+        const newTagIds = entity.tags
+          .filter((tag: TagRequest) => typeof tag.id === "number")
+          .map((tag: TagRequest) => tag.id);
+        const allReviewTags = await PgReviewTag.findAll({
+          where: { review_id: id },
+          transaction: t,
+        });
+        const droppedTags = allReviewTags
+          .map((reviewTag: PgReviewTag) => reviewTag.tag_id)
+          .filter((oldId: number) => !newTagIds.includes(oldId));
+        await Promise.all(
+          droppedTags.map(async (tagId: number) => {
+            const dbTag = await PgTag.findByPk(tagId, { transaction: t });
+            if (!dbTag) {
+              throw new Error(`Tag id ${dbTag} not found`);
+            }
+            reviewToUpdate.$remove("tags", dbTag, { transaction: t });
+          }),
+        );
+
+        // update tags
+        await Promise.all(
```

Comment:
might be wrong here, but do we need this Promise.all() if we are using await in all cases inside the async function? 

`
entity.tags.map(async (tag: TagRequest) => {
  ... a bunch of code which is always awaited...
});
`

Will that not just do the same thing as the Promise.all(). If I understand it correctly, Promise.all() just waits for the all the promises inside to resolve but since you are already using await, the map will always contain the awaited result of the promise. If you remove the await, then the Promise.all() might make more sense?